include/ipfs/client.h

```diff
@@ -377,6 +377,48 @@ class Client {
       /** [in] Key name (local, user-friendly name for the key). */
       const std::string& key_name);
 
+  /** Publish an IPNS name attached to a given value.
+   *
+   * Implements
+   * https://github.com/ipfs/interface-ipfs-core/tree/master/SPEC/NAME.md#name.publish.
+   *
+   * An example usage:
+   * @snippet name.cc ipfs::Client::NamePublish
+   *
+   * @throw std::exception if any error occurs
+   *
+   * @since version 0.4.0 */
+  void NamePublish(
+      /** [in] Id (multihash) of the object to publish. */
+      const std::string& object_id,
+      /** [in] Name of the key to use. This is the local,
+       * human-friendly keyname */
+      const std::string& key_name,
+      /** [out] IPNS name Id (multihash) of the named object. */
+      std::string* name_id,
+      /** [in] Lifetime duration of the record. */
+      const std::string& lifetime = "24h",
```

Comment:
The `resolve` option is missing.

What about passing a `Json` object for all of the 4 options described in https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/NAME.md#namepublish? If a property is not present in the supplied `Json` object, then we don't pass it to curl.

Reply:
Yes, that may be a better API. Should it come before, or after the returned value? If after, then it can be given a default value. If before, then not... I;; push a change that places it before, but without a default.